src/components/Notifications/NoticeBoard.tsx

```diff
@@ -1,62 +1,167 @@
+import { useQuery } from "@tanstack/react-query";
+import { useState } from "react";
+import React from "react";
 import { useTranslation } from "react-i18next";
 
 import CareIcon from "@/CAREUI/icons/CareIcon";
 
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+
+import { Avatar } from "@/components/Common/Avatar";
 import Loading from "@/components/Common/Loading";
 import Page from "@/components/Common/Page";
 
 import routes from "@/Utils/request/api";
-import useTanStackQueryInstead from "@/Utils/request/useQuery";
-import { formatDateTime, formatName } from "@/Utils/utils";
+import query from "@/Utils/request/query";
+import { formatName, relativeTime } from "@/Utils/utils";
+
+import { NotificationData } from "./models";
 
 export const NoticeBoard = () => {
   const { t } = useTranslation();
-  const { data, loading } = useTanStackQueryInstead(routes.getNotifications, {
-    query: { offset: 0, event: "MESSAGE", medium_sent: "SYSTEM" },
+  const { data, isLoading } = useQuery({
+    queryKey: ["notices"],
+    queryFn: query(routes.getNotifications, {
+      queryParams: { offset: 0, event: "MESSAGE", medium_sent: "SYSTEM" },
+    }),
```

Comment:
we would only be showing 1 page (i.e. 14 items) right?

Reply:
So shouldn't need the offset parameter then.